logfire-api/src/logfire_api/__init__.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+try:
+    import logfire
+
+    logfire_installed = True
+except ImportError:
+    logfire_installed = False
+
+
+if TYPE_CHECKING:
+
+    class LogfireConfig: ...
+
+    GLOBAL_CONFIG = LogfireConfig()
+
+    class Logfire:
+        def __init__(
+            self,
+            *,
+            config: LogfireConfig = GLOBAL_CONFIG,
+            sample_rate: float | None = None,
+            tags: Sequence[str] = (),
+            console_log: bool = True,
+            otel_scope: str = 'logfire',
+        ) -> None: ...
+
+else:
+
+    class Logfire:
+        def __init__(self, *args: Any, **kwargs: Any) -> None:
+            if logfire_installed:
+                return logfire.Logfire(*args, **kwargs)
```

Comment:
You need to add methods like `span()` which return something functionally consistent with `logfire.span()`.

And also.

```py
def __getattr__(self, attr):
    return MagicMock()
```

We perhaps we shoud also implement all the know methods with a signature of `(*args, **kwargs) -> None`, then leave this as a fallback.

Reply:
Returning a MagicMock outside of tests seems bad IMO. Can we follow the opentelemetry-api way of doing this?